service_streamer/service_streamer.py

```diff
@@ -478,21 +525,30 @@ def send_request(self, task_id, request_id, model_input):
         self._redis.lpush(self._redis_request_queue_name, pickle.dumps(message))
 
     def recv_response(self, timeout):
-        message = self._response_pb.get_message(timeout=timeout)
-        if message:
-            return pickle.loads(message["data"])
+        try:
+            message = self._response_pb.get_message(timeout=timeout)
+            if message:
+                return pickle.loads(message["data"])
+        except RedisError as e:
+            logger.warning('recv_response: redis error occurs in get_message, {}', str(e))
+            self._setup()  # 重新订阅
```

Comment:
There might be an issue here. `self._setup()` fails when redis server is stopped, but when redis server restarts, this `self._setup()` usually will not get executed because the above `try: ` entry now returns with no error. 

```suggestion
            while True:
                try:
                    self._redis.ping()
                    self._setup() # 重新订阅
                    break
                except RedisError as e:
                    time.sleep(0.2)
```

Reply:
I find that completely removal of the `self._setup()` part in the error handling step also works. It looks like redis-py recovers the connection and the previously subscribed channel after the failure.
If the `_setup()` is executed and erred during the redis-server's downtime, then redis-py cannot recover previous subscription anymore.